### PR TITLE
Add Migration package to handle migrations from version x to version y

### PR DESCRIPTION
**Pull request by JohnChangUK**
Add in `Migration` package to handle migrations from version X to version Y.
- This is needed to decouple contract X from contract Y. 

Spoke with @movekevin , this is an appropriate way of decoupling contracts during migration, and having the migration contract object be responsible for calling the migrate functions in each contract.
- Anyone can call the migration contract, which migrates these resources to the updated contract:
  - `CollectionRefs`
  - `CollectionProperties`
  - `TokenRefs`

## Testing
Deployed 3 contracts to testnet:
1. TokenMinter contract: https://explorer.aptoslabs.com/account/0x186be62b3c3e54fa01fcc4d1e147372ab24cf6dd4ec09bbf6d570f094f7f94fe/modules/code/coin_payment?network=testnet
2. TokenMinterV2 contract: https://explorer.aptoslabs.com/account/0xf0995d360365587c500cc171d1416bad10a331b9c71871a1aec5f2c37ff43124/modules/code/coin_payment?network=testnet
3. Migration contract: https://explorer.aptoslabs.com/account/0xd3cb5491ef963455c6a3dab9ec8e8d49eae0cd76b2863813635977c788b07131/modules/code/migration?network=testnet

### CODE REVIEW

File: token-minter/sources/migration/migration_helper.move [hunk 1]
```diff
@@ -0,0 +1,13 @@
+module minter::migration_helper {
+
+    use aptos_framework::object;
+
+    // The seed for the migration contract
+    const MIGRATION_CONTRACT_SEED: vector<u8> = b"migration::migration_contract";
```
Comment (jillxuu) [inline token-minter/sources/migration/migration_helper.move:6]: rename to `minter::migration`?

**Review by jillxuu — APPROVED** [review]
approve to unblock. @JohnChangUK to see whether there's a cleaner way to handle migration, right now each module has extensive lines of migration out (potentially migration in in the future).

File: migration/sources/migration.move [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+    #[test_only]
+    use minter_v2::collection_components_v2;
+    #[test_only]
+    use minter_v2::collection_properties_v2;
+    #[test_only]
+    use minter_v2::token_components_v2;
+
+    /// Migration function used for migrating the refs from one object to another.
+    /// This is called when the contract has been upgraded to a new address and version.
+    /// This function is used to migrate the refs from the old object to the new object.
```
[the remark this answers is not in the document]
Comment (JohnChangUK) [reply migration/sources/migration.move:34]: Will do

File: token-minter/sources/modules/collection_components.move [hunk 3]
```diff
@@ -22,6 +23,8 @@ module minter::collection_components {
     const EFIELD_NOT_MUTABLE: u64 = 3;
     /// The collection does not have ExtendRef, so it is not extendable.
     const ECOLLECTION_NOT_EXTENDABLE: u64 = 4;
+    /// Not the migration object signer.
```
[the remark this answers is not in the document]
Comment (JohnChangUK) [reply token-minter/sources/modules/collection_components.move:26]: sounds good, will change

File: token-minter/sources/migration/migration_helper.move (hunk 1 above)
Comment (JohnChangUK) [reply token-minter/sources/migration/migration_helper.move:6]: Sounds good

**Pull request merged**